stdlib/SparseArrays/src/higherorderfns.jl

```diff
@@ -1079,8 +1127,10 @@ broadcast(f::Tf, A::SparseMatrixCSC, ::Type{T}) where {Tf,T} = broadcast(x -> f(
 
 function copy(bc::Broadcasted{PromoteToSparse})
     bcf = flatten(bc)
-    if is_supported_sparse_broadcast(bcf.args...)
-        broadcast(bcf.f, map(_sparsifystructured, bcf.args)...)
+    if is_specialcase_sparse_broadcast(bcf.f, bcf.args...)
+        return _copy(bcf.f, bcf.args...)
+    elseif is_supported_sparse_broadcast(bcf.args...)
+        return broadcast(bcf.f, map(_sparsifystructured, bcf.args)...)
```

Comment:
It's pretty awful that `_sparsifystructured` copies transposed sparse vectors into a new CSC matrix.  If we didn't do that (and goodness we really shouldn't — but that'll take much bigger changes), then you could just fall through to this branch.  Maybe we should name things in a way that makes this restriction/reasoning a bit more clear.  Perhaps something like:

```julia
	     if can_skip_sparsification(bcf.f, bcf.args...)
	        return _copy(bcf.f, bcf.args...)
	    elseif is_supported_sparse_broadcast(bcf.args...)
	        return broadcast(bcf.f, map(_sparsifystructured, bcf.args)...)
```

Unrelatedly, it's confusing that we typically call/implement `_copy` but here we call `broadcast`.  Looking through this again, I think this `broadcast` call here should really be `_copy` — using `broadcast` adds yet more indirection and more work for Julia and almost certainly ends right back at `_copy` anyhow… and if it doesn't I think it'd be a recursive stack overflow.  But maybe I'm forgetting something...

Reply:
Doing some testing with Revise.jl interactively suggests no problem with changing `broadcast` to `_copy`. I've started rebuilding the system image and will try running a couple of the test suites as well.

Reply:
`return broadcast` -> `return _copy` in the last commit.